File: api/src/middleware/cors.js
export const corsMiddleware = async (context) => {
  const { request } = context;
  
  // Get the origin from the request
  const origin = request.headers.get('Origin') || '*';
  
  // Define allowed origins
  const allowedOrigins = [
    'http://localhost:3000',
    'http://localhost:8787',
    'http://127.0.0.1:3000',
    'http://127.0.0.1:8787'
  ];
  
  // Check if the origin is allowed - for development, allow all origins
  const isAllowedOrigin = true; // allowedOrigins.includes(origin) || origin === '*';
  
  // Set the appropriate CORS origin
  const corsOrigin = isAllowedOrigin ? origin : allowedOrigins[0];
  
  // Create CORS headers
  const corsHeaders = {
    'Access-Control-Allow-Origin': corsOrigin,
    'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
    'Access-Control-Allow-Headers': 'Content-Type, Authorization, X-Requested-With',
    'Access-Control-Allow-Credentials': 'true',
  };
  
  // Handle preflight requests
  if (request.method === 'OPTIONS') {
    return new Response(null, {
      status: 204,
      headers: {
        ...corsHeaders,
        'Access-Control-Max-Age': '86400',
      }
    });
  }
  
  // Add CORS headers to all responses
  context.corsHeaders = corsHeaders;
}; 

File: fastapi-backend/middleware/cors.py
"""
CORS Middleware

Migrated from api/src/middleware/cors.js to provide the same CORS behavior
in the FastAPI backend using built-in FastAPI CORS middleware.
"""

from fastapi.middleware.cors import CORSMiddleware as FastAPICORSMiddleware
import logging

from config.settings import settings

logger = logging.getLogger(__name__)

def setup_cors_middleware(app):
    """Setup CORS middleware for the FastAPI app"""
    
    # Define allowed origins
    allowed_origins = [
        'http://localhost:3000',
        'http://localhost:8787', 
        'http://127.0.0.1:3000',
        'http://127.0.0.1:8787'
    ]
    
    # For development, allow all origins
    if settings.ENVIRONMENT == "development":
        allowed_origins = ["*"]
    
    app.add_middleware(
        FastAPICORSMiddleware,
        allow_origins=allowed_origins,
        allow_credentials=True,
        allow_methods=["GET", "POST", "PUT", "DELETE", "OPTIONS"],
        allow_headers=["Content-Type", "Authorization", "X-Requested-With"],
        max_age=86400,
    )
    
    logger.info("CORS middleware configured")
    logger.info(f"Allowed origins: {allowed_origins}")
    logger.info("CORS allows all origins in development mode")